packages/react/src/components/CommandCenter/create.tsx

```diff
@@ -0,0 +1,146 @@
+            value={inputStore.value}
+            onChange={handleInputChange}
+            placeholder={inputStore.placeholder}
+            label={inputStore.label}
+        />;
+    });
+
+    return CommandCenterInput;
+}
+
+const createCommandCenterSuggestions = ({ suggestionsStore, onSuggestionSelect }: {
+    suggestionsStore: SuggestionsStore,
+    onSuggestionSelect: (suggestion: Suggestion) => void
+}) => {
+    const setHighlightedSuggestionCommand = (command: Suggestion) => {
+        suggestionsStore.highlightedSuggestion = command;
+    }
+
+    const CommandCenterSuggestions = observer(() => {
+        useEffect(() => {
+            suggestionsStore.init();
+            return () => {
+                suggestionsStore.dispose();
+            };
+        }, []);
+
+        if (!suggestionsStore.suggestions?.length) return null;
+
+        return <CommandCenterSuggestionsComponent
+            setHighlightedSuggestionCommand={setHighlightedSuggestionCommand}
+            suggestions={suggestionsStore.suggestions}
+            onSuggestionClick={onSuggestionSelect}
+            highlightedSuggestionCommand={suggestionsStore.highlightedSuggestion}
+        />;
+    });
+
+    return CommandCenterSuggestions;
+}
+
+export type CommandCenterProps = {
+    fetchSuggestions: (input: string, projectContext?: ProjectContextType) => Promise<Suggestion[]>;
+}
+
+export const createCommandCenter = ({
+    fetchSuggestions,
+}: CommandCenterProps) => {
+    const inputStore = new InputStore();
+    const suggestionsStore = new SuggestionsStore();
+    const commandCenterStore = new CommandCenterStore(
+        inputStore,
+        suggestionsStore,
+        fetchSuggestions
+    );
+
+    const onSuggestionSelect = (suggestion?: Suggestion) => {
+        if (!suggestion) return;
+        commandCenterStore.handleSuggestion(suggestion);
+    }
+
+    const handleClose = () => {
+        commandCenterStore.isOpen = false;
+    }
+
+    const CommandCenterSuggestions = createCommandCenterSuggestions({ suggestionsStore, onSuggestionSelect });
+
+    const CommandCenterInput = createCommandCenterInput({ suggestionsStore, inputStore, onSuggestionSelect });
+
+    const CommandCenter = observer(() => {
+        useEffect(() => {
+            commandCenterStore.init();
+            return () => {
+                commandCenterStore.dispose();
+            };
+        }, []);
+
+        // Syncing between react context and mobx store. 
+        // Wouldn't be needed if we used a global store for the project context data. 
+        const projectContext = useProjectContext();
+        useEffect(() => {
+            commandCenterStore.syncProjectContext(projectContext);
+        }, [projectContext]);
+
+        return <CommandCenterComponent
+            isOpen={commandCenterStore.isOpen}
+            onClose={handleClose}
+            CommandCenterSuggestions={CommandCenterSuggestions}
+            CommandCenterInput={CommandCenterInput}
+        />
```

Comment:
We use Dependency Injection to inject pre-build components into CommandCenterComponent. 
This means the CommandCenterComponent is only responsible for it's own rendering and CSS.

This eventually helps in: 
- Avoiding prop drilling for props that'd be needed downstream
- Unncessary re-renders due to props being passed to children.
- Separation of concerns. The CommandCenterInput can essentially be treated as an individual component which would be easier to change/manipulate when needed.
- Cleaner separation of UI layer and business logic layers.
- Each unit is individually test-able.